src/popup/components/AccountInfo.vue

```diff
@@ -105,10 +118,25 @@ export default defineComponent({
         : '',
     );
 
+    watch(
+      () => [props.isChainName, props.account.address],
+      async ([isChainName, address]) => {
+        if (isChainName && address) {
+          const middleware = await getMiddleware();
+          resolvedChainName.value = (await middleware.getName(address)).name;
+        } else {
+          resolvedChainName.value = '';
+        }
+        isLoading.value = false;
+      },
+      { immediate: true },
+    );
```

Comment:
I strongly suggest to replace this logic with `getName` method taken from `aeNames` composable.

Reply:
I created a new `getChainName` that uses the middleware to resolve a name hash to a name address